src/android/SecureStorage.java

```diff
@@ -150,15 +158,15 @@ public void run() {
             if (value != null) {
                 JSONObject json = new JSONObject(value);
                 final byte[] encKey = Base64.decode(json.getString("key"), Base64.DEFAULT);
-                JSONObject data = json.getJSONObject("value");
+                final JSONObject data = json.getJSONObject("value");
                 final byte[] ct = Base64.decode(data.getString("ct"), Base64.DEFAULT);
                 final byte[] iv = Base64.decode(data.getString("iv"), Base64.DEFAULT);
                 final byte[] adata = Base64.decode(data.getString("adata"), Base64.DEFAULT);
                 cordova.getThreadPool().execute(new Runnable() {
                     public void run() {
                         try {
                             byte[] decryptedKey = rsa.decrypt(encKey, service2alias(service));
-                            String decrypted = new String(AES.decrypt(ct, decryptedKey, iv, adata));
+                            String decrypted = new String(AES.decrypt(ct, decryptedKey, iv, adata, data.getString("mode")));
```

Comment:
I have a feeling that this isn't documented in the README.md, correct?

Reply:
I are right :) This comes from another pull request, i merged. Can you add it by yourself, if you have the time. I am quite busy this week. Thanks !

Reply:
In our case, `data.getString("mode")` will always be `null`, so the default value (AES.GLOBAL_CIPHER = `CCM`) will be used.

The last argument `cipherMode` of function `AES.decrypt()` has been added in order be able to change the value (later), in the `set` API.

Reply:
ok, i add some documentation, and an option to choose the AES cypher mode, when setting a value (Android only).